FindCACTUS.cmake

```diff
@@ -1,145 +1,232 @@
 #.rst:
+#
 # FindCACTUS
 # ----------
 #
-# Finds CACTUS include directory and libraries
+# This module can be used to find libraries provided by CACTUS. It can be used
+# using the standard
+# `find_package <https://cmake.org/cmake/help/latest/command/find_package.html>`_
+# function. It is possible to specify the list of required libraries using the
+# `COMPONENTS` and `OPTIONAL_COMPONENTS` arguments of `find_package` (all
+# components are required if nothing is specified). The following libraries are
+# supported:
+#
+# * `uhal_uhal`
+# * `amc13_amc13`
+# * `amc13_tools`
+#
+# A target of the form `CACTUS::<lib>` is created for every library.
+# Dependencies are handled automatically.
+#
+# The following variables are set by this module:
+#
+# ::
+#
+#     CACTUS_FOUND          -- If CACTUS was found
+#
+#     CACTUS_INCLUDE_DIRS   -- CACTUS include directories
+#     CACTUS_<lib>_LIBARRY  -- Location of the 'lib' library
+#
+# ..note::
+#
+#     Version checking is not supported because no version information is
+#     present in the CACTUS headers.
+#
 
 include(CMakeParseArguments)
+include(FindPackageHandleStandardArgs)
+
+# Exported variables
+set(CACTUS_FOUND TRUE)
 
 # List of all supported libs
-set(cactus_all_libs "")
+set(CACTUS_all_libs "")
 
 # Declares a library that can be required using the COMPONENTS argument
-function(cactus_library name)
-    cmake_parse_arguments(ARG "THREADS" "HEADER" "DEPENDS" "" ${ARGN})
+macro(_cactus_library name)
+    cmake_parse_arguments(ARG "NO_SONAME" "HEADER" "DEPENDS" ${ARGN})
 
-    set(cactus_${name}_header ${ARG_HEADER} PARENT_SCOPE)
-    set(cactus_${name}_depends ${ARG_DEPENDS} PARENT_SCOPE)
+    set(cactus_${name}_depends ${ARG_DEPENDS})
+    set(cactus_${name}_nosoname ${ARG_NO_SONAME})
     list(APPEND cactus_all_libs ${name})
-    set(cactus_all_libs "${cactus_all_libs}" PARENT_SCOPE)
-endfunction()
+endmacro()
 
 # List all supported libs and their dependencies
-cactus_library(uhal_uhal HEADER "uhal/uhal.hpp")
-cactus_library(amc13_amc13 HEADER "amc13/AMC13Simple.hh")
-cactus_library(amc13_tools HEADER "amc13/AMC13.hh" DEPENDS amc13_amc13)
+_cactus_library(uhal_uhal HEADER "uhal/uhal.hpp")
+_cactus_library(amc13_amc13 HEADER "amc13/AMC13Simple.hh")
```

Comment:
The two AMC13 libraries do not have their SONAME set, so I think the option should be added.